controllers/azureidentity_controller_test.go

```diff
@@ -0,0 +1,132 @@
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-azure/util/system"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	capifeature "sigs.k8s.io/cluster-api/feature"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestAzureIdentityControllerReconcileAzureCluster(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+	req := ctrl.Request{NamespacedName: types.NamespacedName{Name: "fake-cluster", Namespace: "fake-ns"}}
+	scheme := runtime.NewScheme()
+	err := infrav1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+	err = aadpodv1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	azureCluster := &infrav1.AzureCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-cluster",
+			Namespace: "fake-ns",
+		},
+	}
+
+	bindings := &aadpodv1.AzureIdentityBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-binding",
+			Namespace: system.GetManagerNamespace(),
+			Labels: map[string]string{
+				clusterv1.ClusterNameLabel:    "another-fake-cluster",
+				infrav1.ClusterLabelNamespace: "fake-ns",
+			},
+		},
+		Spec: aadpodv1.AzureIdentityBindingSpec{
+			AzureIdentity: "managedIdentity",
+		},
+	}
+
+	azIdentity := &aadpodv1.AzureIdentity{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "managedIdentity",
+			Namespace: system.GetManagerNamespace(),
+		},
+	}
+
+	client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(azureCluster, bindings, azIdentity).Build()
+	aiRec := &AzureIdentityReconciler{
+		Client:           client,
+		Recorder:         record.NewFakeRecorder(42),
+		ReconcileTimeout: 20 * time.Second,
+		WatchFilterValue: "fake",
+	}
+	_, err = aiRec.Reconcile(ctx, req)
+	g.Expect(err).ToNot(HaveOccurred())
+}
+
+func TestAzureIdentityControllerReconcileAzureManagedControlPlane(t *testing.T) {
+	g := NewWithT(t)
+	defer featuregatetesting.SetFeatureGateDuringTest(t, capifeature.Gates, capifeature.MachinePool, true)()
+	ctx := context.Background()
+	req := ctrl.Request{NamespacedName: types.NamespacedName{Name: "fake-controlplane", Namespace: "fake-ns"}}
+	scheme := runtime.NewScheme()
+	err := infrav1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+	err = aadpodv1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+	azureManagedControlPlane := &infrav1.AzureManagedControlPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-controlplane",
+			Namespace: "fake-ns",
+		},
+	}
+	bindings := &aadpodv1.AzureIdentityBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-binding",
+			Namespace: system.GetManagerNamespace(),
+			Labels: map[string]string{
+				clusterv1.ClusterNameLabel:    "another-fake-cluster",
+				infrav1.ClusterLabelNamespace: "fake-ns",
+			},
+		},
+		Spec: aadpodv1.AzureIdentityBindingSpec{
+			AzureIdentity: "managedIdentity",
+		},
+	}
+	azIdentity := &aadpodv1.AzureIdentity{
```

Comment:
Looks like the same bindings and azIdentity are used across the two tests. I think it would be cleaner to re-use them in some way. An example is creating a helper function like this: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/controllers/azuremachine_controller_test.go#L510

Reply:
I'll make the function being this is being duplicated.

Reply:
Pushed on the latest update. Resolving.